### PR TITLE
Rename method typed.javadsl.FishingOutcomes.continue  due to compilation error in Java (#29951)

### PR DESCRIPTION
**Pull request by Captain1653**
References #29951 

I didn't find any mentions of the method in the docs. Did I miss something?

### CODE REVIEW

**Comment by He-Pin** [comment]
Make it depracated instead of remove？

**Comment by Captain1653** [comment]
I'm not sure.  This method wasn't compiled ever. I think it can be renamed and added to MimaExcludes.

**Comment by Captain1653** [comment]
Hm... I added mima exclude, but I still have the error 
`[error] akka-actor-testkit-typed: Failed binary compatibility check against com.typesafe.akka:akka-actor-testkit-typed_2.12:2.6.18! Found 2 potential problems
[error]  * method continue()akka.actor.testkit.typed.FishingOutcome in object akka.actor.testkit.typed.javadsl.FishingOutcomes does not have a correspondent in current version
[error]    filter with: ProblemFilters.exclude[DirectMissingMethodProblem]("akka.actor.testkit.typed.javadsl.FishingOutcomes.continue")
[error]  * static method continue()akka.actor.testkit.typed.FishingOutcome in class akka.actor.testkit.typed.javadsl.FishingOutcomes does not have a correspondent in current version
[error]    filter with: ProblemFilters.exclude[DirectMissingMethodProblem]("akka.actor.testkit.typed.javadsl.FishingOutcomes.continue")`

File: akka-actor-testkit-typed/src/main/mima-filters/2.6.20.backwards.exclude/rename-fishing-outcomes-continue-java.excludes [hunk 1]
```diff
@@ -0,0 +1,2 @@
+# FishingOutcomes.continue() cannot be used in Java code due to reserved word, see https://github.com/akka/akka/issues/29951
+ProblemFilters.exclude[DirectMissingMethodProblem]("akka.actor.testkit.typed.javadsl.FishingOutcomes.continue")
```
Comment (johanandren) [inline akka-actor-testkit-typed/src/main/mima-filters/2.6.20.backwards.exclude/rename-fishing-outcomes-continue-java.excludes:2]: Wrong directory, should be 2.6.19 as in binary changes against latest released version.

Comment (Captain1653) [reply akka-actor-testkit-typed/src/main/mima-filters/2.6.20.backwards.exclude/rename-fishing-outcomes-continue-java.excludes]: Thank you @johanandren . I thought about new release :) . I squashed commits for better history.

File: akka-actor-testkit-typed/src/main/scala/akka/actor/testkit/typed/javadsl/TestProbe.scala [hunk 2]
```diff
@@ -24,7 +24,7 @@ object FishingOutcomes {
   /**
    * Consume this message and continue with the next
    */
-  def continue(): FishingOutcome = FishingOutcome.Continue
```
Comment (He-Pin) [inline akka-actor-testkit-typed/src/main/scala/akka/actor/testkit/typed/javadsl/TestProbe.scala:27]: Should we add a new method `continueAndCollect` and deprecated the old `continue` one?

Comment (Captain1653) [reply akka-actor-testkit-typed/src/main/scala/akka/actor/testkit/typed/javadsl/TestProbe.scala:27]: Let's wait for other reviewers, ok ?

**Review by raboof — APPROVED** [review]
I think we can skip deprecating the old `continue`, since this is the javadsl of the testkit, and Java (or Kotlin) code using it cannot have existed (since it wouldn't compile).

**Review by johanandren — APPROVED** [review]

**Comment by Captain1653** [comment]
@johanandren Is it worth to close the original issue #29951 and add milesone?

**Pull request merged**